Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/Merchant/Overlay.hs

```diff
@@ -50,22 +51,22 @@ cacheOverlayForCity merchantOperatingCityId cfg = do
 makeMerchantOpCityIdKey :: Id MerchantOperatingCity -> Text
 makeMerchantOpCityIdKey id = "driver-offer:CachedQueries:Overlay:MerchantOperatingCityId-" <> id.getId
 
-findByMerchantOpCityIdPNKeyLangaugeUdf :: (CacheFlow m r, EsqDBFlow m r) => Id MerchantOperatingCity -> Text -> Language -> Maybe Text -> m (Maybe Overlay)
-findByMerchantOpCityIdPNKeyLangaugeUdf id pnKey language udf1 =
-  Hedis.safeGet (makeMerchantIdPNKeyLangaugeUdf id pnKey language udf1) >>= \case
+findByMerchantOpCityIdPNKeyLangaugeUdfVehicleCategory :: (CacheFlow m r, EsqDBFlow m r) => Id MerchantOperatingCity -> Text -> Language -> Maybe Text -> Maybe DVC.VehicleCategory -> m (Maybe Overlay)
```

Comment:
We had to do same fallback as MerchantMessage where vehicleVariant is Nothing here as well.